task-maker-format/src/ioi/sanity_checks/task.rs

```diff
@@ -57,3 +60,53 @@ impl SanityCheck<IOITask> for BrokenSymlinks {
         Ok(())
     }
 }
+
+/// Check that cpp source files don't contain #include <bits/stdc++.h> (or whitespace variants of it)
+#[derive(Debug, Default)]
+pub struct NoBitsStdCpp;
+
+impl SanityCheck<IOITask> for NoBitsStdCpp {
+    fn name(&self) -> &'static str {
+        "NoBitsStdCpp"
+    }
+
+    fn pre_hook(&mut self, task: &IOITask, eval: &mut EvaluationData) -> Result<(), Error> {
+        lazy_static! {
+            static ref RE: Regex =
+                Regex::new(r###"(?m)^#\s*include\s*(?:<|")bits/stdc\+\+\.h(?:>|").*$"###)
+                    .expect("Invalid regex");
+        }
+
+        for att in list_files(
+            &task.path,
+            vec![
+                "att/*.cpp",
+                "att/*.cc",
+                "sol/grader.cpp",
+                "sol/grader.cc",
+                "sol/stub.cpp",
+                "sol/stub.cc",
```

Comment:
I'm not sure we want to bother the problem setters with this. They may want to still use `bits/stdc++.h` and don't care about compilation times. To me, the sense of this check is to prevent the contestants to fail to compile _the attachments_ when they are not using gcc.

I will think about this more thoroughly.

Reply:
The idea behind this was to save the extra parsing time when evaluating solutions during a contest, since they are linked with the grader in `sol`. It's probably not that significant, but may help if there's bursts of submissions. If you prefer removing this check I'm ok with that, the important part was to check that the attachments don't contain it.